### PR TITLE
[#6473] Main server clean up (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Rebased work done by @june-releford on main.

Running through testing environment.

### CODE REVIEW

File: server/core/src/initServer.cpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (korydraughn) [inline server/core/src/initServer.cpp:1]: remove

File: server/main_server/src/rodsServer.cpp [hunk 2]
```diff
@@ -190,8 +184,7 @@ namespace
 
         const int num_hex_bytes = 2 * num_random_bytes;
         std::snprintf(buf, num_hex_bytes + 1, "%s", ss.str().c_str());
-
-        return 0;
+
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:187]: remove

File: server/main_server/src/rodsServer.cpp [hunk 3]
```diff
@@ -869,6 +852,153 @@ namespace
...
+			agentProc_t *tmpAgentProc = *agentProcHead;
+			while ( tmpAgentProc->next != NULL ) {
+				tmpAgentProc = tmpAgentProc->next;
+			}
+			tmpAgentProc->next = agentProc;
+			agentProc->next = NULL;
+		}
+		return 0;
+	}
+
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:1001]: remove

File: server/main_server/src/rodsServer.cpp [hunk 4]
```diff
@@ -2028,50 +2180,77 @@ agentProc_t* getConnReqFromQueue()
...
+        while (SpawnReqHead) {
+            agentProc_t* mySpawnReq = SpawnReqHead;
+            SpawnReqHead = SpawnReqHead->next;
+
+            spwn_req_lock.unlock();
+            auto status = spawnAgent(mySpawnReq, &ConnectedAgentHead);
+            close(mySpawnReq->sock);
+
+            if (status < 0) {
+                rodsLog(LOG_NOTICE,
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:2229]: log_server::info

File: server/main_server/src/rodsServer.cpp [hunk 5]
```diff
@@ -2028,50 +2180,77 @@ agentProc_t* getConnReqFromQueue()
...
+                        "spawnAgent error for puser=%s and cuser=%s from %s, stat=%d",
+                        mySpawnReq->startupPack.proxyUser,
+                        mySpawnReq->startupPack.clientUser,
+                        inet_ntoa(mySpawnReq->remoteAddr.sin_addr),
+                        status);
+                std::free(mySpawnReq);
+            }
+            else {
+                rodsLog(LOG_DEBUG,
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:2238]: log_server::info

File: server/re/src/sharedmemory.cpp [hunk 6]
```diff
@@ -49,6 +49,7 @@ unsigned char *prepareServerSharedMemory( const std::string& _key ) {
     }
 }
 
+// TODO(june): this has been used other places, but it does nothing?
```
Comment (korydraughn) [inline server/re/src/sharedmemory.cpp:52]: Remove this function if not used.

**Review by korydraughn — COMMENTED** [review]
Don't forget to run `git clang-format`.

File: server/core/include/irods/rodsConnect.h [hunk 7]
```diff
@@ -18,17 +18,9 @@
 #define INITIAL_NOT_DONE                0
 #define INITIAL_DONE                    1
 
+// TODO(june): maybe move to rodsServer.hpp?
 #define LOCK_FILE_PURGE_TIME            7200 // purge lock files every 2 hr.
```
Comment (alanking) [inline server/core/include/irods/rodsConnect.h:22]: Seems reasonable to move this...

File: server/re/src/sharedmemory.cpp (hunk 6 above)
Comment (alanking) [inline server/re/src/sharedmemory.cpp:52]: Torch?

**Review by alanking — COMMENTED** [review]
Nice. Couple of TODOs milling about.

File: server/re/src/sharedmemory.cpp (hunk 6 above)
Comment (alanking) [reply server/re/src/sharedmemory.cpp:52]: Thanks, GitHub.

File: server/core/src/initServer.cpp (hunk 1 above)
Comment (korydraughn) [reply server/core/src/initServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 2 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 3 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 4 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 5 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/core/include/irods/rodsConnect.h (hunk 7 above)
Comment (korydraughn) [reply server/core/include/irods/rodsConnect.h]: done

**Comment by korydraughn** [comment]
All unit tests pass.

Running core tests now.

**Comment by korydraughn** [comment]
All core tests pass.

Test Time: 2 hr 24 mins

**Comment by korydraughn** [comment]
Some of the log messages are incorrect.

I'm going to handle those before we merge this.

File: server/core/src/initServer.cpp [hunk 8]
```diff
@@ -1057,116 +937,50 @@ setRsCommFromRodsEnv( rsComm_t *rsComm ) {
...
-    }
-
-    if ( *agentProcHead == NULL ) {
-        *agentProcHead = agentProc;
-        agentProc->next = NULL;
-        return 0;
-    }
+void close_all_l1_descriptors(RsComm& _comm)
+{
+    rodsLog(LOG_DEBUG, "[%s:%d] Closing all L1 descriptors ...", __func__, __LINE__);
```
Comment (korydraughn) [inline server/core/src/initServer.cpp:942]: log_agent::debug

File: server/main_server/src/rodsServer.cpp [hunk 9]
```diff
@@ -869,6 +852,142 @@ namespace
...
     } // daemonize
+
+    void print_usage(const char* prog)
+    {
+        printf("Usage: %s [-uvVqs]\n", prog);
+        printf(" -u  user command level, remain attached to the tty (foreground)\n");
+        printf(" -v  verbose (LOG_NOTICE)\n");
+        printf(" -V  very verbose (LOG_DEBUG10)\n");
+        printf(" -q  quiet (LOG_ERROR)\n");
+        printf(" -s  log SQL commands\n");
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:863]: std::printf / std::cout / fmt::print

File: server/main_server/src/rodsServer.cpp [hunk 10]
```diff
@@ -1169,19 +1285,38 @@ int serverMain(const bool enable_test_mode = false, const bool write_to_stdout =
...
-            return status;
+        // Start read worker thread
+        for (int i = 0; i < NUM_READ_WORKER_THR; ++i) {
+            try {
+                ReadWorkerThread[i] = new boost::thread(readWorkerTask);
+            }
+            catch (const boost::thread_resource_error&) {
+                log_server::error(
+                    "boost encountered a thread_resource_error during readWorkerTask thread construction in "
+                    "serverMain.");
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:1296]: remove serverMain

File: server/main_server/src/rodsServer.cpp [hunk 11]
```diff
@@ -1169,19 +1285,38 @@ int serverMain(const bool enable_test_mode = false, const bool write_to_stdout =
...
         }
 
+        // Start spawn manager thread
+        try {
+            SpawnManagerThread = new boost::thread(task_spawn_manager);
+        }
+        catch (const boost::thread_resource_error&) {
+            log_server::error(
+                "boost encountered a thread_resource_error during task_spawn_manager thread construction in "
+                "serverMain.");
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:1308]: remove serverMain

File: server/main_server/src/rodsServer.cpp [hunk 12]
```diff
@@ -1169,19 +1285,38 @@ int serverMain(const bool enable_test_mode = false, const bool write_to_stdout =
...
         if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
             try {
-                PurgeLockFileThread = new boost::thread(purgeLockFileWorkerTask);
+                PurgeLockFileThread = new boost::thread(task_purge_lock_file);
             }
             catch (const boost::thread_resource_error&) {
-                log_server::error(
-                    "boost encountered a thread_resource_error during thread construction in serverMain.");
+                log_server::error("boost encountered a thread_resource_error during task_purge_lock_file thread "
+                                  "construction in serverMain.");
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:1319]: no serverMain

File: server/main_server/src/rodsServer.cpp [hunk 13]
```diff
@@ -1296,7 +1431,29 @@ int serverMain(const bool enable_test_mode = false, const bool write_to_stdout =
...
-        stopProcConnReqThreads();
+
+        // Stop spawn manager thread
+        SpawnReqCond.notify_all();
+        try {
+            SpawnManagerThread->join();
+        }
+        catch (const boost::thread_resource_error&) {
+            log_server::error(
+                "boost encountered a thread_resource_error during spawn manager thread join in serverMain.");
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:1442]: no serverMain

File: server/main_server/src/rodsServer.cpp [hunk 14]
```diff
@@ -1296,7 +1431,29 @@ int serverMain(const bool enable_test_mode = false, const bool write_to_stdout =
...
+        // Stop read worker threads
+        for (int i = 0; i < NUM_READ_WORKER_THR; ++i) {
+            ReadReqCond.notify_all();
+
+            try {
+                ReadWorkerThread[i]->join();
+            }
+            catch (const boost::thread_resource_error&) {
+                log_server::error(
+                    "boost encountered a thread_resource_error during read worker thread join in serverMain.");
```
Comment (korydraughn) [inline server/main_server/src/rodsServer.cpp:1454]: no serverMain

File: server/core/src/initServer.cpp (hunk 8 above)
Comment (korydraughn) [reply server/core/src/initServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 9 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 10 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 11 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 12 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 13 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

File: server/main_server/src/rodsServer.cpp (hunk 14 above)
Comment (korydraughn) [reply server/main_server/src/rodsServer.cpp]: done

**Comment by korydraughn** [comment]
I've verified that the most recent changes are safe (ran unit tests one more time and they all pased).

Any objections to adding the pounds?

**Review by alanking — APPROVED** [review]

**Comment by trel** [comment]
all good from me...  why three commits rather than one?

**Comment by trel** [comment]
well, or at least, one for june, one for you.

**Comment by korydraughn** [comment]
The original PR included three commits. I only added one to fix issues.

**Comment by trel** [comment]
fair enough.   # em up if you're happy.

**Comment by korydraughn** [comment]
Will squash and add the pounds if things check out.

**Review by SwooshyCueb — APPROVED** [review]

**Comment by korydraughn** [comment]
Squashed and pounded.

**Pull request merged**